python/lsst/pipe/tasks/makeDirectWarp.py

```diff
@@ -0,0 +1,799 @@
+        dtype=Warper.ConfigClass,
+    )
+    maskedFractionWarper = ConfigField(
+        doc="Configuration for the warp that warps the mask fraction image",
+        dtype=Warper.ConfigClass,
+    )
+    coaddPsf = ConfigField(
+        doc="Configuration for CoaddPsf",
+        dtype=CoaddPsfConfig,
+    )
+    idGenerator = DetectorVisitIdGeneratorConfig.make_field()
+
+    def setDefaults(self) -> None:
+        super().setDefaults()
+        self.warper.warpingKernelName = "lanczos3"
+        self.maskedFractionWarper.warpingKernelName = "bilinear"
+        self.prewarpInterpolation.badMaskPlanes = self.badMaskPlanes
+
+    def validate(self) -> None:
+        # Allow for subset below.
+        if not set(self.badMaskPlanes) == set(self.prewarpInterpolation.badMaskPlanes):
+            raise FieldValidationError(
+                self.__class__.badMaskPlanes,
+                self,
+                "The badMaskPlanes and prewarpInterpolation.bad_mask_planes must be the same.",
+            )
+        super().validate()
+
+
+class MakeDirectWarpTask(PipelineTask):
+    """Warp calexps (and PVIs) onto a common projection.
+
+    This task iterates over multiple calexps (corresponding to different
+    detectors) from a single visit that overlap the target image. Pixels that
+    receive no input from any calexp are set to NaN in the output image, and
+    NO_DATA bit is set in the mask plane.
+
+    This differs from the standard `MakeWarp` Task in the following
+    ways:
+
+    1. No selection on ccds at the time of warping. This is done later during
+       the coaddition stage.
+    2. Interpolate over a set of masked pixels before warping.
+    3. Generate an image where each pixel denotes how much of the pixel is
+       masked.
+    4. Generate multiple noise warps with the same interpolation applied.
+    5. No option to produce a PSF-matched warp.
+    """
+
+    ConfigClass = MakeDirectWarpConfig
+    _DefaultName = "makeDirectWarp"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("inputRecorder")
+        self.makeSubtask("prewarpInterpolation")
+
+        self.warper = Warper.fromConfig(self.config.warper)
+        self.maskedFractionWarper = Warper.fromConfig(self.config.maskedFractionWarper)
+
+    @inheritDoc(PipelineTask)
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # Docstring to be augmented with info from PipelineTask.runQuantum
+        """Notes
+        -----
+        Obtain the list of input detectors from calexp_list.  Sort them by
+        detector order (to ensure reproducibility).  Then ensure all input
+        lists are in the same sorted detector order.
+        """
+        detectorOrder = [
+            ref.datasetRef.dataId["detector"] for ref in inputRefs.calexp_list
+        ]
+        detectorOrder.sort()
+        inputRefs = reorderRefs(inputRefs, detectorOrder, dataIdKey="detector")
```

Comment:
I'm not sure if this is really needed. Maybe @laurenam has some history behind this when she introduced it in `makeWarp.py`.

Reply:
Sorry, just seeing this now.  Indeed, this was part of the gen2-gen3-revealed non-derterministic sorting of those days (the sorting was added in commit https://github.com/lsst/pipe_tasks/commit/775f0afb0da81f99b48c8a01c2a6015bd0fd5255).  If the quantum are all now sorted going into tasks, I guess this is no longer necessary (but I have no insight into Jim's caveat...)